test/complex_test.rb

```diff
@@ -13,56 +14,95 @@ def initialize
-      allow(Yast::Stage)
-        .to receive(:initial)
-        .and_return(false)
-    end
+    context "when not running in installer" do
+      before(:each) do
+        allow(Yast::Stage)
+          .to receive(:initial)
+          .and_return(false)
+      end
 
-    it "returns true for input different than :abort" do
-      expect(subject.input_done?(:no_abort)).to eql true
-    end
+      it "returns true for input different than :abort" do
+        expect(subject.input_done?(:no_abort)).to eql true
+      end
+
+      it "returns true for input equal to :abort in case of no user modifications" do
+        allow(Yast::LanItems)
+          .to receive(:GetModified)
+          .and_return(false)
 
-    it "returns true for input equal to :abort in case of no user modifications" do
-      allow(Yast::LanItems)
-        .to receive(:GetModified)
-        .and_return(false)
+        expect(subject.input_done?(:abort)).to eql true
+      end
 
-      expect(subject.input_done?(:abort)).to eql true
+      it "asks user for abort confirmation for input equal to :abort and user did modifications" do
+        allow(Yast::LanItems)
+          .to receive(:GetModified)
+          .and_return(true)
+
+        expect(subject)
+          .to receive(:ReallyAbort)
+          .and_return(BOOLEAN_PLACEHOLDER)
+
+        expect(subject.input_done?(:abort)).to eql BOOLEAN_PLACEHOLDER
+      end
     end
 
-    it "asks user for abort confirmation for input equal to :abort and user did modifications" do
-      allow(Yast::LanItems)
-        .to receive(:GetModified)
-        .and_return(true)
+    context "when running in installer" do
+      before(:each) do
+        allow(Yast::Stage)
+          .to receive(:initial)
+          .and_return(true)
+      end
 
-      expect(subject)
-        .to receive(:ReallyAbort)
-        .and_return(BOOLEAN_PLACEHOLDER)
+      it "asks user for installation abort confirmation for input equal to :abort" do
+        expect(Yast::Popup)
+          .to receive(:ConfirmAbort)
+          .and_return(BOOLEAN_PLACEHOLDER)
 
-      expect(subject.input_done?(:abort)).to eql BOOLEAN_PLACEHOLDER
+        expect(subject.input_done?(:abort)).to eql BOOLEAN_PLACEHOLDER
+      end
     end
   end
 
-  context "when running in installer" do
-    before(:each) do
-      allow(Yast::Stage)
-        .to receive(:initial)
-        .and_return(true)
+  describe "#DeviceProtocol" do
+    let(:ipaddr) { "192.168.0.120" }
+    let(:managed) { { "STARTMODE" => "managed" } }
+    let(:static) { { "BOOTPROTO" => "static", "IPADDR" => ipaddr } }
+    let(:empty) { { "BOOTPROTO" => "", "IPADDR" => ipaddr } }
+    let(:no_bootproto) { { "IPADDR" => ipaddr } }
+    # IPADDR are set just to show that returns the protocol instead of the IP
+    let(:dhcp) { { "BOOTPROTO" => "dhcp", "IPADDR" => ipaddr } }
+    let(:none) { { "BOOTPROTO" => "none", "IPADDR" => ipaddr } }
+    let(:zero) { { "BOOTPROTO" => "static", "IPADDR" => ipaddr } }
```

Comment:
true, i copied and pasted to added to use it with 0.0.0.0 address but just used merge at the end, so I will remove or fix it. Thnx